src/compiler/checker.ts

```diff
@@ -12604,6 +12861,14 @@ namespace ts {
                 return result;
             }
 
+            // Do not report any error if we are doing so for stateless function component as such error will be error will be handle in "resolveCustomJsxElementAttributesType".

```

Comment:
The reason that JSX stateless function can't be treat as normal call because

- To be valid stateless function, we have to check that the return type match JSXElementType (see :`defaultTryGetJsxStatelessFunctionAttributesType`) I don't want to move that check here since it is such JSX specific. 

- When trying to resolve JSX tag-name that is not intrinsic, we do following : try as if it is stateless, if fails then try as stateful.